### PR TITLE
add an output modifier for get bulk secret api

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

For the HTTP Get Bulk Secret API, the output in master has changed from 
```
{
    "randomKey": {
        "randomKey": "value"
    },
    "redisPassword": {
        "redisPassword": "root123"
    }
}
```
to
```
{
    "data": {
        "randomKey": {
            "secrets": {
                "randomKey": "value"
            }
        },
        "redisPassword": {
            "secrets": {
                "redisPassword": "root123"
            }
        }
    }
}
```

This PR fixes this issue. Adds a unit test for the same. 

## Issue reference

Related to #6301 
Related to #6207 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by ItalyPaleAle — COMMENTED** [review]
Can you add an e2e test please so this won’t happen again?

**Comment by mukundansundar** [comment]
> Can you add an e2e test please so this won’t happen again?

agree E2E is needed. 
But to catch this situation, the Unit test has been modified to specifically look for a particular response and not only test for status success.

**Comment by mukundansundar** [comment]
will add E2E by tomorrow and then this can be merged.

**Comment by mukundansundar** [comment]
moving to draft to add E2E

File: tests/config/dapr_tests_cluster_role_binding.yaml [hunk 1]
```diff
@@ -21,5 +21,5 @@ metadata:
 rules:
 - apiGroups: [""]
   resources: ["secrets"]
-  verbs: ["get"]
-  
+  verbs: ["get", "list"]
```
Comment (mukundansundar) [inline tests/config/dapr_tests_cluster_role_binding.yaml:24]: Based on https://docs.dapr.io/operations/components/component-secrets/#non-default-namespaces

**Review by artursouza — APPROVED** [review]

**Pull request merged**